### PR TITLE
[Feature tore] Add a test for scheduled runs to spark system  tests

### PR DESCRIPTION
**Pull request by katyakats**

### CODE REVIEW

File: tests/system/feature_store/test_spark_engine.py [hunk 1]
```diff
@@ -80,3 +86,108 @@ def test_basic_remote_spark_ingest(self):
...
         )
+
+    @pytest.mark.parametrize("partitioned", [True, False])
+    def test_schedule_on_filtered_by_time(self, partitioned):
+        name = f"sched-time-{str(partitioned)}"
+
+        now = datetime.now() + timedelta(minutes=2)
+
+        path = "v3io:///bigdata/bla.parquet"
+        fsys = fsspec.filesystem(v3iofs.fs.V3ioFS.protocol)
```
Comment (urihoenig) [inline tests/system/feature_store/test_spark_engine.py:97]: Any reason not to use store_manager.get_or_create_store & get_filesystem?

File: tests/system/feature_store/test_spark_engine.py [hunk 2]
```diff
@@ -80,3 +86,108 @@ def test_basic_remote_spark_ingest(self):
             spark_context=self.spark_service,
             run_config=fs.RunConfig(local=False),
         )
+
+    @pytest.mark.parametrize("partitioned", [True, False])
+    def test_schedule_on_filtered_by_time(self, partitioned):
+        name = f"sched-time-{str(partitioned)}"
```
Comment (urihoenig) [inline tests/system/feature_store/test_spark_engine.py:92]: Isn't it the same name as in the non-spark tests? Maybe it's better to add "spark" to the name?

Comment (urihoenig) [reply tests/system/feature_store/test_spark_engine.py:92]: Actually, it's a different project, no need to rename.
Is it the same code? Maybe we can add a utils file and import it from there in both test files? (Not critical)

File: tests/system/feature_store/test_spark_engine.py [hunk 3]
```diff
@@ -80,3 +86,108 @@ def test_basic_remote_spark_ingest(self):
...
+                ],
+                "first_name": ["moshe", "dina", "katya", "uri"],
+                "data": [50, 10, 25, 30],
+            }
+        ).to_parquet(path=path)
+
+        sleep(60)
+        resp = svc.get(
+            [
+                {"first_name": "yosi"},
```
Comment (urihoenig) [inline tests/system/feature_store/test_spark_engine.py:175]: Yossi

File: mlrun/feature_store/api.py [hunk 4]
```diff
@@ -684,9 +684,11 @@ def _ingest_with_spark(
                 df.write.mode("append").save(**spark_options)
             target.set_resource(featureset)
             target.update_resource_status("ready")
-        max_time = df.agg({timestamp_key: "max"}).collect()[0][0]
-        for target in featureset.status.targets:
-            featureset.status.update_last_written_for_target(target.path, max_time)
+
+        if isinstance(source, BaseSourceDriver) and source.schedule:
```
[the remark this answers is not in the document]
Comment (katyakats) [reply mlrun/feature_store/api.py:688]: we calculated max_time till now with no real reason in non-scheduled spark ingestion

File: tests/system/feature_store/test_spark_engine.py (hunk 1 above)
Comment (katyakats) [reply tests/system/feature_store/test_spark_engine.py:97]: not really

**Review by urihoenig — APPROVED** [review]

**Review by Hedingber — APPROVED** [review]

**Pull request merged**